templates/rhsm-katello-reconfigure.erb

```diff
@@ -106,13 +106,19 @@ fi
   service goferd status >/dev/null && \
   service goferd restart >/dev/null 2&>1
 
+[ -f /etc/init.d/goferd ] && \
 [ -f /bin/systemctl ] && \
   systemctl try-restart goferd >/dev/null 2>&1
 
-
-FQDN=`hostname -f`
-if [ $? == "0" ] && [ "$FQDN" != "localhost" ] && [ -d /etc/rhsm/facts/ ]; then
-  echo "{\"network.hostname-override\":\"$FQDN\"}" > /etc/rhsm/facts/katello.facts
+# If RHSM version is less than 1.18.1-1, we have to update the hostname-override fact, since those versions don't
+# have the network.fqdn fact. This doesn't apply to EL5 systems, where we always want to add the hostname-override
+if (test -f /etc/redhat-release && grep -q -i "Red Hat Enterprise Linux Server release 5" /etc/redhat-release) || \
+   (test -f /etc/centos-release && grep -q -i "CentOS Linux release 5" /etc/centos-release) || \
+   test ${RHSM_VERSION[0]:-0} -lt 1 -o ${RHSM_VERSION[1]:-0} -lt 18 -o \( ${RHSM_VERSION[1]:-0} -eq 18 -a ${RHSM_VERSION[2]:-0} -lt 2 \); then
```

Comment:
This is slightly terrifying and knowing it's already used in another place doesn't soothe me.

Reply:
We discussed converting this to python, I think that would be a good long term goal and would make complicated logic easier